### PR TITLE
Update kraken to v1.1.1

### PR DESCRIPTION
**Pull request by dfornika**
:information_source:
If this is your first pull request to Bioconda please ping `@bioconda/core` so it can be reviewed and to request being added as a member of the Bioconda team. Members of the Bioconda team are able to merge their own pull requests once tests and linting have passed.

----------------

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by epruesse** [comment]
@biocondabot recheck

**Comment by epruesse** [comment]
This is weird. It didn't build anything.

**Comment by epruesse** [comment]
Alright. That's fixed.

**Comment by dfornika** [comment]
I'm seeing:

```
make: g++: Command not found
```

...in the build log.

**Comment by epruesse** [comment]
Yes. Thought that was listed.  Does it need just g++ oder both?

**Comment by HadrienG** [comment]
`circleci build` works on my machine, but the test fails online with the following:

```
/Users/distiller/project/miniconda/conda-bld/kraken_1557123069873/_build_env/bin/../include/c++/v1/__threading_support:156:1: error: unknown type name 'mach_port_t'
mach_port_t __libcpp_thread_get_port();
^
/Users/distiller/project/miniconda/conda-bld/kraken_1557123069873/_build_env/bin/../include/c++/v1/__threading_support:300:1: error: unknown type name 'mach_port_t'
mach_port_t __libcpp_thread_get_port() {
^
/Users/distiller/project/miniconda/conda-bld/kraken_1557123069873/_build_env/bin/../include/c++/v1/__threading_support:301:12: error: use of undeclared identifier 'pthread_mach_thread_np'
    return pthread_mach_thread_np(pthread_self());
           ^
3 errors generated.
```


Googling around I've found [this](https://boringssl.googlesource.com/boringssl/+/63a0797ff247f13870b649c3f6239d80be202752)

Could it be that it fails because circleCI's online service uses an older Xcode?
Do we have a fix for this?

File: recipes/kraken/meta.yaml [hunk 1]
```diff
@@ -23,11 +23,20 @@ build:
...
-    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - llvm-openmp  # [osx]
 
+  host:
+    - perl
+    - clangdev  # [osx]
+    - llvm-openmp  # [osx]
   run:
     - jellyfish 1.*
```
Comment (epruesse) [inline recipes/kraken/meta.yaml]: This probably needs to be `bioconda::jellyfish` as there is a conflicting package in conda-forge.

**Comment by epruesse** [comment]
> Could it be that it fails because circleCI's online service uses an older Xcode?

Quite possibly. I ran into that myself at some point.

File: recipes/kraken/meta.yaml (hunk 1 above)
Comment (dfornika) [reply recipes/kraken/meta.yaml]: I've changed it to `bioconda::jellyfish 1.*`

File: recipes/kraken/meta.yaml [hunk 2]
```diff
@@ -23,11 +23,20 @@ build:
...
+    - perl
+    - clangdev  # [osx]
+    - llvm-openmp  # [osx]
   run:
-    - jellyfish 1.*
+    - bioconda::jellyfish 1.*
     - perl
+    - rsync
+    - wget
+    - libcxx  # [osx]
```
Comment (epruesse) [inline recipes/kraken/meta.yaml]: Was this one necessary? It should not be...

**Comment by epruesse** [comment]
@dfornika I'll trigger the merge tomorrow, after looking this over once more. Looks good. Only the libc++ OSX dependency looks unusual.

File: recipes/kraken/meta.yaml (hunk 2 above)
Comment (HadrienG) [reply recipes/kraken/meta.yaml]: it did not find openmp without it. we noticed the same thing for [kraken2](https://github.com/bioconda/bioconda-recipes/pull/14971)

**Comment by druvus** [comment]
@dfornika are you planning to also bump krakenuniq?

File: recipes/kraken/meta.yaml (hunk 2 above)
Comment (HadrienG) [reply recipes/kraken/meta.yaml]: mmmh, my bad, it seems to build without it.

**Comment by dfornika** [comment]
@druvus I wasn't planning on it. I'm not closely involved in these projects so I'm not exactly sure how they depend on or interact with each other.

I do see that krakenuniq has a v0.5.8 release but it's not on anaconda.org so I could get a PR started if nobody else beats me to it.

**Comment by epruesse** [comment]
@biocondabot recheck

**Comment by epruesse** [comment]
@bioconda/core What do we do about Jellyfish? It works because CF Jellyfish is 0.6.1, but as soon as they upgrade to 1.x this will be broken. The `bioconda::jellyfish` is lost on the way into the `repodata.json` and has no effect.

**Comment by dpryan79** [comment]
The conda-forge maintainers haven't responded as far as I can tell. I guess we kick this can down the road until it's a problem or they respond :(

**Comment by HadrienG** [comment]
They have responded haven't they? Is it ok to merge this?

**Comment by epruesse** [comment]
It's going to take time. We'll have to fix a bunch of dependencies, one more won't change things much.
@BiocondaBot  merge

**Pull request merged**